src/helpers/messages/send_message.ts

```diff
@@ -93,6 +96,60 @@ export async function sendMessage(
     }
   }
 
+  if (content.components?.length) {
+    let actionRowCounter = 0;
+
+    for (const component of content.components) {
+      // 5 Link buttons can not have a custom_id
+      if (isButton(component)) {
+        if (
+          component.type === ButtonStyles.Link &&
+          component.customId
+        ) {
+          throw new Error(Errors.LINK_BUTTON_CANNOT_HAVE_CUSTOM_ID);
+        }
+        if (
+          !component.customId
+        ) {
+          if (component.type !== ButtonStyles.Link) {
+            throw new Error(Errors.BUTTON_REQUIRES_CUSTOM_ID);
+          }
+        }
```

Comment:
Little bit less nesting
```suggestion
        if (component.type === ButtonStyles.Link) {
          if (component.customId) throw new Error(Errors.LINK_BUTTON_CANNOT_HAVE_CUSTOM_ID);
        } else if (!component.customId) throw new Error(Errors.BUTTON_REQUIRES_CUSTOM_ID);
```